src/api/tests/service/learn/test_learn_funcs.py

```
@@ -0,0 +1,312 @@
+import sys
+import types
+import unittest
+import unittest.mock
+
+from flask import Flask
+from flask_sqlalchemy import SQLAlchemy
+
+# Provide a lightweight Redis stub if the dependency is missing in the test env.
+try:
+    import redis as _redis  # type: ignore  # noqa: F401
```

Ruff (0.14.14)

[warning] Unused `noqa` directive (non-enabled: `F401`)

Remove unused `noqa` directive

(RUF100)